### PR TITLE
Typed persistence tests: RecoveryPermitterSpec

### PR DESCRIPTION
**Pull request by kelebra**
This PR partially addresses issue #24687. To be exact: port of test `akka.persistence.RecoveryPermitterSpec`.

I tried to keep this one as 'typed' as I could however as I needed direct 'permitter' interaction and messages from different protocols, it ended up like that.

Suggestions and improvements are welcome, tried to keep test structure as close to the untyped one as possible to simplify review.

### CODE REVIEW

File: akka-persistence-typed/src/test/scala/akka/persistence/typed/internal/RecoveryPermitterSpec.scala [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+import akka.testkit.typed.scaladsl.{ ActorTestKit, TestProbe }
+import com.typesafe.config.{ Config, ConfigFactory }
+import org.scalatest.concurrent.Eventually
+
+import scala.concurrent.duration._
+import scala.util.control.NoStackTrace
+
+object RecoveryPermitterSpec {
+
+  class TestExc extends RuntimeException("simulated exc") with NoStackTrace
```
Comment (ktoso) [inline akka-persistence-typed/src/test/scala/akka/persistence/typed/internal/RecoveryPermitterSpec.scala]: our usual pattern for this is `TE` ("Test Exception") with a message "Boom!" :-)

**Review by ktoso — APPROVED** [review]
LGTM, thanks

**Comment by ktoso** [comment]
Structure looks fine, and it covers the permit obtaining in the new behavior, looks good to merge. Would be great to continue porting the other tests from here onwards

File: akka-persistence-typed/src/test/scala/akka/persistence/typed/internal/RecoveryPermitterSpec.scala (hunk 1 above)
Comment (kelebra) [reply akka-persistence-typed/src/test/scala/akka/persistence/typed/internal/RecoveryPermitterSpec.scala]: Fixed :)

**Comment by kelebra** [comment]
Yep, that was the plan. Will try to do separate PRs per test case for 'reviewability'. Should we trigger CI?

**Comment by ktoso** [comment]
OK TO TEST

**Comment by ktoso** [comment]
Yeah separate PRs per test will be nice, thanks

**Comment by ktoso** [comment]
Marked with milestone as it's a 1/n part of the above issue

File: akka-persistence-typed/src/test/scala/akka/persistence/typed/internal/RecoveryPermitterSpec.scala [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+import akka.testkit.typed.scaladsl.{ ActorTestKit, TestProbe }
+import com.typesafe.config.{ Config, ConfigFactory }
+import org.scalatest.concurrent.Eventually
+
+import scala.concurrent.duration._
+import scala.util.control.NoStackTrace
+
+object RecoveryPermitterSpec {
+
+  class TE extends RuntimeException("Boom!") with NoStackTrace
```
Comment (patriknw) [inline akka-persistence-typed/src/test/scala/akka/persistence/typed/internal/RecoveryPermitterSpec.scala]: could have used `akka.testkit.typed.TE`

**Pull request merged**